daffodil-io/src/main/scala/org/apache/daffodil/io/DataOutputStream.scala

```diff
@@ -96,6 +97,15 @@ trait DataOutputStream extends DataStreamCommon
 
   def maybeAbsBitPos0b: MaybeULong
 
+  /**
+   * These values are used for output streams that could change from
+   * ByteArray's to File based output streams.
+   */
+  def chunkSizeInBytes: Int
+  def maxBufferSizeInBytes: Long
+  def tempFilePath: Path
```

Comment:
Maybe ``tempDirPath`` to make it clear this is a path to a directory that temp files will be created in?